include/SDL3/SDL_gpu.h

```diff
@@ -1675,6 +1675,36 @@ typedef struct SDL_GPUStorageTextureWriteOnlyBinding
 
 /* Device */
 
+/**
+ * Checks for GPU runtime support.
+ *
+ * \param format_flags a bitflag indicating which shader formats the app is
+ *                     able to provide.
+ * \param name the preferred GPU driver, or NULL to let SDL pick the optimal
+ *             driver.
+ * \returns SDL_TRUE if supported, SDL_FALSE otherwise.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_CreateGPUDevice
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_QueryGPUSupport(
+    SDL_GPUShaderFormat format_flags,
+    const char *name);
+
+/**
+ * Checks for GPU runtime support.
+ *
+ * \param props the properties to use.
+ * \returns SDL_TRUE if supported, SDL_FALSE otherwise.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_CreateGPUDeviceWithProperties
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_QueryGPUSupportWithProperties(
```

Comment:
What properties are intended to be used here?

Reply:
Forgot to add to the docs: These are the same properties used with CreateGPUDevice.

Reply:
So is this SDL_QueryGPUSupportsProperties()?

Reply:
That name could work - we can fix this as part of #10825.

Reply:
We can drop the Query to match other SDL question functions, so it could just be SDL_GPUSupportsProperties() and SDL_GPUSupportsFormats().

Reply:
Done via https://github.com/libsdl-org/SDL/commit/16ff7503b7baac7f92f2274d73cca944f0a97337